compiler/dyno/lib/resolution/scope-queries.cpp

```diff
@@ -885,17 +884,18 @@ doResolveUseStmt(Context* context, const Use* use,
     const Scope* foundScope = findUseImportTarget(context, scope, r,
                                                   expr, VIS_USE, oldName);
     if (foundScope != nullptr) {
-      // First, add VisibilitySymbols entry for the symbol itself
-      if (newName.isEmpty()) {
-        r->addVisibilityClause(foundScope, VisibilitySymbols::SYMBOL_ONLY,
-                               isPrivate, convertOneName(oldName));
-      } else {
+      // First, add VisibilitySymbols entry for the symbol itself,
+      // iff the use is renamed or non-public
```

Comment:
This comment would be more useful if it explained why the non-public condition is there. My guess is that it's because when searching for symbols, we will continue through `public use`'s, so there's no point in keeping the symbol in this `r`, but if it's private, we'll stop here?

Reply:
Even mentioning that it's a spec would not be pretty nice in my opinion. Even something as simple as "according to the spec, private uses create a scope, so add VisibilitySymbols entry for the symbol".